File: Python3/1779. Find Nearest Point That Has the Same X or Y Coordinate.py
class Solution:
    def nearestValidPoint(self, x: int, y: int, points: List[List[int]]) -> int:
        
        minn = inf
        dis = {}
        
        for i, (a, b) in enumerate(points):
            if a == x or b == y:
                d = abs(a-x + b-y)
                if d not in dis:
                    dis[d] = i
                minn = min(minn, d)
        
        return dis[minn] if minn != inf else -1

File: Java/1779. Find Nearest Point That Has the Same X or Y Coordinate.java
class Solution {
    public int nearestValidPoint(int x, int y, int[][] points) {
        int min = Integer.MAX_VALUE;
        Map<Integer, Integer> dis = new HashMap<>();
        int d;
        
        for (int i=0; i<points.length; i++) {
            if (points[i][0] == x || points[i][1] == y) {
                d = Math.abs(points[i][0] - x + points[i][1] - y);
                if (!dis.containsKey(d)) {
                    dis.put(d, i);
                    min = Math.min(min, d);
                }
            }
        }
        
        return min == Integer.MAX_VALUE ? -1 : dis.get(min);
    }
}